src/Umbraco.Core/PropertyEditors/ValueConverters/ImageCropperValue.cs

```diff
@@ -59,77 +61,72 @@ public ImageCropperCrop GetCrop(string alias)
                 : Crops.FirstOrDefault(x => x.Alias.InvariantEquals(alias));
         }
 
-        internal void AppendCropBaseUrl(StringBuilder url, ImageCropperCrop crop, bool defaultCrop, bool preferFocalPoint)
+        internal ImageUrlGenerationOptions GetCropBaseOptions(string url, ImageCropperCrop crop, bool defaultCrop, bool preferFocalPoint)
         {
             if (preferFocalPoint && HasFocalPoint()
                 || crop != null && crop.Coordinates == null && HasFocalPoint()
                 || defaultCrop && HasFocalPoint())
             {
-                url.Append("?center=");
-                url.Append(FocalPoint.Top.ToString(CultureInfo.InvariantCulture));
-                url.Append(",");
-                url.Append(FocalPoint.Left.ToString(CultureInfo.InvariantCulture));
-                url.Append("&mode=crop");
+                return new ImageUrlGenerationOptions { ImageUrl = url, FocalPoint = new ImageUrlGenerationOptions.FocalPointPosition { Left = FocalPoint.Left, Top = FocalPoint.Top } };
             }
             else if (crop != null && crop.Coordinates != null && preferFocalPoint == false)
             {
-                url.Append("?crop=");
-                url.Append(crop.Coordinates.X1.ToString(CultureInfo.InvariantCulture)).Append(",");
-                url.Append(crop.Coordinates.Y1.ToString(CultureInfo.InvariantCulture)).Append(",");
-                url.Append(crop.Coordinates.X2.ToString(CultureInfo.InvariantCulture)).Append(",");
-                url.Append(crop.Coordinates.Y2.ToString(CultureInfo.InvariantCulture));
-                url.Append("&cropmode=percentage");
+                return new ImageUrlGenerationOptions { ImageUrl = url, Crop = new ImageUrlGenerationOptions.CropCoordinates { X1 = crop.Coordinates.X1, X2 = crop.Coordinates.X2, Y1 = crop.Coordinates.Y1, Y2 = crop.Coordinates.Y2 } };
             }
             else
             {
-                url.Append("?anchor=center");
-                url.Append("&mode=crop");
+                return new ImageUrlGenerationOptions { ImageUrl = url, DefaultCrop = true };
             }
         }
 
         /// <summary>
         /// Gets the value image url for a specified crop.
         /// </summary>
-        public string GetCropUrl(string alias, bool useCropDimensions = true, bool useFocalPoint = false, string cacheBusterValue = null)
+        [Obsolete("Use the overload that takes an IImageUrlGenerator")]
+        public string GetCropUrl(string alias, bool useCropDimensions = true, bool useFocalPoint = false, string cacheBusterValue = null) => GetCropUrl(alias, Current.ImageUrlGenerator, useCropDimensions, useFocalPoint, cacheBusterValue);
+
+        /// <summary>
+        /// Gets the value image url for a specified crop.
+        /// </summary>
+        public string GetCropUrl(string alias, IImageUrlGenerator imageUrlGenerator, bool useCropDimensions = true, bool useFocalPoint = false, string cacheBusterValue = null)
```

Comment:
Is any reason to insert ImageurlGenator instead of using internal variable, as we register that as Unique, it means we probably could use one ImageUrlGenator, instead of allowing inject few of them in the same time?

Reply:
I'm not sure what you mean by this.  The ImageCropperValue class is a model class either manually constructed or deserialized from JSON.  Since we can't pass it in through the constructor due to the deserialization, and we don't want a static reference for injection purposes, we need to pass this in to the method.

Reply:
I didn't notice it is class which is deserialized from JSON my comment is not really relevant in that case :)